src/Proxy.hs

```diff
@@ -520,60 +520,61 @@ getRewardPeriodLength lfb = do
-                            "stakedAmount" .= asiStakedAmount,
-                            "restakeEarnings" .= asiStakeEarnings,
-                            "bakerId" .= _bakerIdentity asiBakerInfo,
-                            "bakerElectionVerifyKey" .= _bakerElectionVerifyKey asiBakerInfo,
-                            "bakerSignatureVerifyKey" .= _bakerSignatureVerifyKey asiBakerInfo,
-                            "bakerAggregationVerifyKey" .= _bakerAggregationVerifyKey asiBakerInfo
-                          ]
-                          <> maybe [] (\bpi -> ["bakerPoolInfo" .= bpi]) asiPoolInfo
-                    case asiPendingChange of
-                      NoChange -> Left . Just $ object $ balanceInfo <> ["accountBaker" .= object infoWithoutPending]
-                      _ -> let bi rpl = object $ infoWithoutPending <> pendingChangeToJSON nextPayday epochDuration rpl asiPendingChange
-                          in Right $ \rpl -> object (balanceInfo <> ["accountBaker" .= bi rpl])
-                  AccountStakingDelegated{..} -> do
-                    let infoWithoutPending = [
-                            "stakedAmount" .= asiStakedAmount,
-                            "restakeEarnings" .= asiStakeEarnings,
-                            "delegationTarget" .= asiDelegationTarget
-                          ]
-                    case asiDelegationPendingChange of
-                      NoChange -> Left . Just $ object $ balanceInfo <> ["accountDelegation" .= object infoWithoutPending]
-                      _ -> let di rpl = object $ infoWithoutPending <> pendingChangeToJSON nextPayday epochDuration rpl asiDelegationPendingChange
-                          in Right $ \rpl -> object (balanceInfo ++ ["accountDelegation" .= di rpl])
-          lastFinBalComp = getBal lastFinInfo
-          bestBalComp = getBal bestInfo
-      let response lastFinBal bestBal = do
-            $(logInfo) $ "Retrieved account balance for " <> addrText
-                        <> ": finalizedBalance=" <> (Text.pack $ show lastFinBal)
-                        <> ", currentBalance=" <> (Text.pack $ show bestBal)
-            sendResponse $ object $ (maybe [] (\b -> ["finalizedBalance" .= b]) lastFinBal) <>
-                              (maybe [] (\b -> ["currentBalance" .= b]) bestBal)
-      case (lastFinBalComp, bestBalComp) of
-        (Left lastFinBal, Left bestBal) -> response lastFinBal bestBal
-        (Left lastFinBal, Right bestBalF) -> runGRPC (getRewardPeriodLength lastFinBlock) $ \rpl -> response lastFinBal (Just (bestBalF rpl))
-        (Right lastFinBalF, Left bestBal) -> runGRPC (getRewardPeriodLength lastFinBlock) $ \rpl -> response (Just (lastFinBalF rpl)) bestBal
-        (Right lastFinBalF, Right bestBalF) -> runGRPC (getRewardPeriodLength lastFinBlock) $ \rpl -> response (Just (lastFinBalF rpl)) (Just (bestBalF rpl))
+    runGRPC (doGetBal accAddr) $ \case
+      Nothing -> sendResponse (object []) -- send an empty object if something
```

Comment:
The only change in lines 523-575 is the case for matching on the introduced `Maybe`, the comments, and the indentation, no?

Reply:
Yes. You can actually hide whitespace changes in github. There is a settings wheel on the top left in the review panel
![Screenshot from 2023-04-20 21-43-31](https://user-images.githubusercontent.com/105212/233471721-61cb981e-6c59-484b-91b4-0c4edb4f733a.png)

Reply:
Very neat! That will come in handy 👍